Few questions I want to answer by analyzing the data I have compiled for a few months of working for Instacart until current day (April 25, 2022)
- How much money have I earned in total?
- How many miles have I driven?
- How many hours have I actively worked?
- What is the highest amount I have earned in a single day?

In [15]:
# Imports
import pandas as pd
import numpy as np
import datetime as dt
from datetime import timedelta

- Read in the instacart data csv and parsed the date column to convert it into datetime data type.
- Read in first 5 rows

In [2]:
# Read in the Instacart CSV containing all of the compiled data
instacart_df = pd.read_csv("Instacart_Analysis_CSV.csv", parse_dates=['Date'])
instacart_df.head()

Date  Batch Earnings   Tips  Adjustments Active Hours  Distance
0 2022-01-22           47.92   9.56        59.97         2:18      31.4
1 2022-01-23          138.99  90.03        60.72        10:00      53.6
2 2022-01-24          108.83  22.12       182.45         5:40      44.2
3 2022-01-25          106.02  35.52       300.23         5:11      35.0
4 2022-01-26           98.27  41.25       361.98         5:54      20.6

In [3]:
instacart_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88 entries, 0 to 87
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Date            88 non-null     datetime64[ns]
 1   Batch Earnings  88 non-null     float64       
 2   Tips            88 non-null     float64       
 3   Adjustments     13 non-null     float64       
 4   Active Hours    88 non-null     object        
 5   Distance        85 non-null     float64       
dtypes: datetime64[ns](1), float64(4), object(1)
memory usage: 4.2+ KB


- Converted 'Active Hours' Column to datetime

In [11]:
instacart_df['Active Hours'] = pd.to_datetime(instacart_df['Active Hours'],format= "%H:%M").dt.time

- Created new column which will sum the amount earned per day.
- Used iloc to parse only the Batch earnings, Tips, and adjustments for the sum.
- Used the insert method to place the new column in a more logical spot in the dataframe.

In [5]:
instacart_df.insert(4, column='Day Total' , value=instacart_df.iloc[:, 1:3].sum(axis=1))

In [6]:
instacart_df.head()

Date  Batch Earnings   Tips  Adjustments  Day Total Active Hours  \
0 2022-01-22           47.92   9.56        59.97      57.48         2:18   
1 2022-01-23          138.99  90.03        60.72     229.02        10:00   
2 2022-01-24          108.83  22.12       182.45     130.95         5:40   
3 2022-01-25          106.02  35.52       300.23     141.54         5:11   
4 2022-01-26           98.27  41.25       361.98     139.52         5:54   

   Distance  
0      31.4  
1      53.6  
2      44.2  
3      35.0  
4      20.6

In [7]:
'''To calculate the total amount of money earned I can just find the sum of the column "Total Amount Earned" as that describes
 the total amount earned per day
'''
total_amount_earned = instacart_df['Day Total'].sum() + instacart_df['Adjustments'].sum() + instacart_df['Distance'].sum() * .30
total_amount_earned


14165.98

Over the past several months I have earned a grand total of $14,165.98 which includes batch earnings, tips, adjustments as well as the gas reimbursement of 30 cents per mile driven

In [8]:
# Total amount of miles driven was given by the previous calculation
total_amount_miles_driven = instacart_df['Distance'].sum()
total_amount_miles_driven

1890.0

Total miles currently driven: 1,890 miles

In [9]:
# Highest single day total earned
max_single_day_earnings = instacart_df['Day Total'].max()
max_single_day_earnings

263.2

In [10]:
# To find the breakdown of this day we can use the loc function and look for the max value's location.
max_earnings_day_location = instacart_df.loc[instacart_df['Day Total'] == max_single_day_earnings]
max_earnings_day_location


Date  Batch Earnings    Tips  Adjustments  Day Total Active Hours  \
8 2022-01-30          150.03  113.17       276.93      263.2         8:59   

   Distance  
8      31.1

The day I earned the most occurred on January 30, 2022. Checking the data for that specific day, I received the largest full service order of $114.35 around 11am and I continued to take orders until just about 9:30pm.